src/main/java/pl/umk/mat/git2befit/controller/workout/TrainingPlanController.java

```diff
@@ -21,54 +18,36 @@
     }
 
     @PostMapping("/generate")
     public ResponseEntity<?> generate(@RequestBody(required = false) TrainingForm trainingForm){
         List<ExerciseExecution> exerciseExecutions = new ArrayList<>();
-        List<Training> trainingPlans = new ArrayList<>();
-        if(trainingForm == null) {
-            for (int i = 1; i <= 9; i++) {
-                if (i % 3 == 0) {
-                    Training training = new Training();
-                    training.setId(11);
-                    training.setId(1);
-                    training.setExercisesExecutions(exerciseExecutions);
-                    exerciseExecutions.clear();
-                    trainingPlans.add(training);
-                }
-                ExerciseExecution exerciseExecution = new ExerciseExecution();
-                exerciseExecution.setExercise(new Exercise(i,
-                        "cwiczenie" + i,
-                        "start" + i,
-                        "execution" + i,
-                        "hints1 /n hint2"
-                        , new ExerciseForm(i, "exerciseForm" + i)
-                        , "videURl" + i, "photo" + i
-                        , new BodyPart(i, "bodyPart" + i)
-                        , List.of(new TrainingType(i, "TrainingType" + i))
-                        , List.of(new Equipment(i, new EquipmentType(i, "kategoria" + i, "urlPhoto" + i)
-                        , "name" + i, "urlVideo" + i)))
-                );
-                exerciseExecution.setCount(8);
-                exerciseExecution.setSeries(3);
-                exerciseExecutions.add(exerciseExecution);
-                System.out.println(exerciseExecution.getExercise().getId());
-
-            }
-        }else {
-            try {
-                trainingPlans = manufacture.createTrainingPlan(trainingForm);
-            } catch (IllegalArgumentException e) {
-                return ResponseEntity.status(HttpStatus.CONFLICT).header("Cause", e.getMessage()).build();
-            }
+        List<Training> trainingPlans;
+        try {
+            trainingPlans = manufacture.createTrainingPlan(trainingForm);
+        } catch (IllegalArgumentException e) {
+            return ResponseEntity.status(HttpStatus.CONFLICT).header("Cause", e.getMessage()).build();
         }
 
         TrainingPlan trainingPlan = new TrainingPlan(trainingForm, trainingPlans);
         return ResponseEntity.ok(trainingPlan);
 
     }
+
+    @PostMapping("/modify/{id}")
+    public ResponseEntity<?> modifyTrainingPlan(Exercise exercise, @RequestBody TrainingForm trainingForm){
+        List<Exercise> similarExercises;
+        try {
+            similarExercises = trainingService.getSimilarExercises(exercise.getId(), trainingForm);
+            return ResponseEntity.ok().body(similarExercises);
+        }catch (IllegalArgumentException e){
+            return ResponseEntity.notFound().header("Cause", e.getMessage()).build();
+        }
+    }
```

Comment:
Tak jak mówiliśmy na code review, wydaje mi się że lepiej byłoby to przenieść do ExerciseController